### PR TITLE
Scalafmt update

### PR DESCRIPTION
**Pull request by poslegm**
I tried to tune config as discussed in https://github.com/scalameta/scalafmt/issues/1952.

Almost all changes are now related to indentation fix in for comprehensions and aligning.

### CODE REVIEW

File: .scalafmt.conf [hunk 1]
```diff
@@ -20,6 +22,7 @@ project.excludeFilters = [
   "core/shared/src/main/scala-2.x/zio/VersionSpecific.scala"
 ]
 
+rewrite.redundantBraces.generalExpressions = false
```
Comment (poslegm) [inline .scalafmt.conf:25]: In scalafmt 2.7.0 if-else braces will be extracted to [rewrite.redundantBraces.ifElseExpressions = false](https://scalameta.org/scalafmt/docs/configuration.html#redundantbraces), so you can enable it again if you want to remove braces around `while` one-liners for example

File: .scalafmt.conf [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
 docstrings = JavaDoc
 lineEndings = preserve
 includeCurlyBraceInSelectChains = false
-danglingParentheses = true
+danglingParentheses.preset = true
 spaces {
   inImportCurlyBraces = true
 }
 optIn.annotationNewlines = true
+newlines.alwaysBeforeMultilineDef = false
```
Comment (poslegm) [inline .scalafmt.conf:15]: A lot of changes could be related to more strict formatting of expressions like this:

```scala
// before
def suite = suite("My suite)(
  testM("My test") { 
    // ...
  }
)

// after
def suite =
  suite("My suite)(
    testM("My test") { 
      // ...
    }
  )
```

I see that first style is preferred in this repo, so disabled it

**Comment by jdegoes** [comment]
@poslegm I like the better alignment in for comprehensions. Can you merge conflicts and re-open? Thank you!

**Comment by poslegm** [comment]
Re-opened with fresh master https://github.com/zio/zio/pull/4106

**Pull request closed without merging**